plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/RewriteStringComparison.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.oracle;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.base.expression.ConnectorExpressionRule;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.QueryParameter;
+import io.trino.plugin.jdbc.expression.ComparisonOperator;
+import io.trino.plugin.jdbc.expression.ParameterizedExpression;
+import io.trino.spi.expression.Call;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.CharType;
+import io.trino.spi.type.VarcharType;
+import oracle.jdbc.OracleTypes;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.argument;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.argumentCount;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.call;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.functionName;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.type;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.variable;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+
+public class RewriteStringComparison
+        implements ConnectorExpressionRule<Call, ParameterizedExpression>
+{
+    private static final Capture<Variable> FIRST_ARGUMENT = newCapture();
+    private static final Capture<Variable> SECOND_ARGUMENT = newCapture();
+    private static final Pattern<Call> PATTERN = call()
+            .with(type().equalTo(BOOLEAN))
+            .with(functionName().matching(Stream.of(ComparisonOperator.values())
+                    .filter(comparison -> comparison != ComparisonOperator.IS_DISTINCT_FROM)
+                    .map(ComparisonOperator::getFunctionName)
+                    .collect(toImmutableSet())
+                    ::contains))
+            .with(argumentCount().equalTo(2))
```

Comment:
nit: extract `type -> type instanceof CharType || type instanceof` to a `isCharacterType(Type type)` method

Reply:
i think `type -> type instanceof CharType || type instanceof VarcharType` is simple enough to read
and this being sibling lines makes it clear we do exact same operation twice

isCharacterType might provoke poking into to check which types it covers